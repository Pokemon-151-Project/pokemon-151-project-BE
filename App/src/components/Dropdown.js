// This component shows different options for the user to display Pokemon based on type, size and height, as well as to reset to default.
// Thanks to w3 for the dropdown source code which I've heavily modified for my purposes here:
// https://www.w3schools.com/howto/howto_css_dropdown.asp

//this is all the different pokemon types to make buttons for
import typesArray from "../utils/Dropdownutils/typesArray";

const Dropdown = (props) => {
  const { changeDisplay } = props;

  return (
    <section className="dropdown">
      {/* Hovering here shows the dropdown menu with display options */}
      <button className="dropbtn" data-testid="dropbtn">
        Display Options (check this out!)
      </button>

      <menu className="dropdown-content">
        <div>
          <div>
            {/* Reset to original display options by refreshing the page*/}
            <button
              className="dropbtn-inner"
              data-testid="reset-btn"
              id="reset-btn"
              onClick={() => {
                changeDisplay(null, null, null, "reset");
              }}
            >
              Reset All
            </button>
          </div>
          <h5
            className="dropdown-display-title"
            data-testid="dropdown-display-title"
          >
            Pick a type:
          </h5>
          {/* Returns 15 buttons, one for each Pokemon type*/}
          {typesArray.map((item) => {
            return (
              <button
                className="dropbtn-inner"
                data-testid="dropbtn-inner"
                id={`${item.toLowerCase()}-btn`}
                key={item}
                onClick={() => {
                  changeDisplay(item);
                }}
              >
                {item}
              </button>
            );
          })}
        </div>

        {/* Begin buttons for sorting pokemon by weight */}
        <div>
          {/* Button to show only small pokemon (<10kg) */}
          <h5 className="dropdown-display-title">Pick a weight:</h5>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="small-btn"
            onClick={() => {
              changeDisplay(null, "small");
            }}
          >
            Small
          </button>

          {/* Button to show only big pokemon (>100kg) */}
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="big-btn"
            onClick={() => {
              changeDisplay(null, "big");
            }}
          >
            Big
          </button>
        </div>

        {/* Begin buttons for sorting pokemon by height */}
        <div>
          <h5 className="dropdown-display-title">Pick a height:</h5>
          {/* Show only short pokemon */}
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="short-btn"
            onClick={() => {
              changeDisplay(null, null, "short");
            }}
          >
            Short
          </button>

          {/* Button to show only tall pokemon */}
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="tall-btn"
            onClick={() => {
              changeDisplay(null, null, "tall");
            }}
          >
            Tall
          </button>
        </div>
      </menu>
    </section>
  );
};

export default Dropdown;
