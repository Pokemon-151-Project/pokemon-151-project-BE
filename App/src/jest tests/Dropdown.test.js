import Dropdown from "../components/Dropdown";
import renderer from "react-test-renderer";

/* eslint-disable testing-library/no-node-access */
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

test("[1] component renders without errors", () => {
  render(<Dropdown />);
});

test("[2] All dropdown buttons render", () => {
  render(<Dropdown />);
  const buttons = screen.queryAllByTestId("dropbtn-inner");
  expect(buttons).toBeTruthy();
  expect(buttons).toHaveLength(19);
});

test("[3] Matches snapshot from 5.8.22", () => {
  const component = renderer.create(<Dropdown />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
