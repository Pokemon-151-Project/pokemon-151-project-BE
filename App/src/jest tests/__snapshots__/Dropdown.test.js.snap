// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[3] Matches snapshot from 5.8.22 1`] = `
<section
  className="dropdown"
>
  <button
    className="dropbtn"
    data-testid="dropbtn"
  >
    Display Options (check this out!)
  </button>
  <menu
    className="dropdown-content"
  >
    <div>
      <div>
        <button
          className="dropbtn-inner"
          data-testid="reset-btn"
          id="reset-btn"
          onClick={[Function]}
        >
          Reset All
        </button>
      </div>
      <h5
        className="dropdown-display-title"
        data-testid="dropdown-display-title"
      >
        Pick a type:
      </h5>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="normal-btn"
        onClick={[Function]}
      >
        Normal
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="fire-btn"
        onClick={[Function]}
      >
        Fire
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="water-btn"
        onClick={[Function]}
      >
        Water
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="electric-btn"
        onClick={[Function]}
      >
        Electric
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="grass-btn"
        onClick={[Function]}
      >
        Grass
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="ice-btn"
        onClick={[Function]}
      >
        Ice
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="fighting-btn"
        onClick={[Function]}
      >
        Fighting
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="poison-btn"
        onClick={[Function]}
      >
        Poison
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="ground-btn"
        onClick={[Function]}
      >
        Ground
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="flying-btn"
        onClick={[Function]}
      >
        Flying
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="psychic-btn"
        onClick={[Function]}
      >
        Psychic
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="bug-btn"
        onClick={[Function]}
      >
        Bug
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="rock-btn"
        onClick={[Function]}
      >
        Rock
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="ghost-btn"
        onClick={[Function]}
      >
        Ghost
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="dragon-btn"
        onClick={[Function]}
      >
        Dragon
      </button>
    </div>
    <div>
      <h5
        className="dropdown-display-title"
      >
        Pick a weight:
      </h5>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="small-btn"
        onClick={[Function]}
      >
        Small
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="big-btn"
        onClick={[Function]}
      >
        Big
      </button>
    </div>
    <div>
      <h5
        className="dropdown-display-title"
      >
        Pick a height:
      </h5>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="short-btn"
        onClick={[Function]}
      >
        Short
      </button>
      <button
        className="dropbtn-inner"
        data-testid="dropbtn-inner"
        id="tall-btn"
        onClick={[Function]}
      >
        Tall
      </button>
    </div>
  </menu>
</section>
`;
